python_modules/dagster/dagster/_core/scheduler/instigation.py

```diff
@@ -403,16 +403,29 @@ def requested_asset_materialization_count(self) -> int:
         return len(asset_partitions)
 
     @property
-    def requested_asset_keys(self) -> AbstractSet[AssetKey]:
+    def requested_assets_and_partitions(self) -> Mapping[AssetKey, AbstractSet[str]]:
         check.invariant(
             self.tick_data.instigator_type == InstigatorType.AUTO_MATERIALIZE,
             "Only auto-materialize ticks set requested_asset_keys",
         )
-        asset_keys = set()
+        partitions_by_asset_key = {}
         for run_request in self.tick_data.run_requests or []:
             for asset_key in run_request.asset_selection or []:
-                asset_keys.add(asset_key)
-        return asset_keys
+                if asset_key not in partitions_by_asset_key:
+                    partitions_by_asset_key[asset_key] = set()
+
+                if run_request.partition_key:
+                    partitions_by_asset_key[asset_key].add(run_request.partition_key)
```

Comment:
Just want to highlight that unpartitioned assets can technically be part of partitioned asset jobs.

AMP will (currently) never create a run with both partitioned and unpartitioned assets, so it's fine for now, but I can imagine worlds in which this invariant stops holding.